src/Microsoft.ComponentDetection.Detectors/go/GoComponentWithReplaceDetector.cs

```diff
@@ -0,0 +1,560 @@
+namespace Microsoft.ComponentDetection.Detectors.Go;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.ComponentDetection.Common;
+using Microsoft.ComponentDetection.Common.Telemetry.Records;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Extensions.Logging;
+using MoreLinq;
+using Newtonsoft.Json;
+
+public class GoComponentWithReplaceDetector : FileComponentDetector, IExperimentalDetector
```

Comment:
I'd prefer that we use an inherited class here rather than a full copy paste of the file. If the main addition is `ReplaceGoComponents` we can override the functions that call it in the inherited class

Reply:
would rather leave it like this since the only reason its in a new files for experiments. The other detector would get deleted and that would mean reconfiguring the code again